fv/wireguard_test.go

```diff
@@ -0,0 +1,442 @@
+
+		It("the Wireguard rule-entry/route-table shouldn't list wireguard rule", func() {
+			for _, felix := range felixes {
+				rule, err := felix.ExecOutput("ip", "rule", "show", "pref", wireguardRoutingRulePriorityDefault)
+				Expect(err).NotTo(HaveOccurred())
+				rule = strings.TrimSpace(rule)
+				Expect(rule).To(MatchRegexp(`\d+:\s+not from all fwmark 0x\d+ lookup \d+`))
+
+				routeTable := routingRuleRegExp.FindString(rule)
+				route, err := felix.ExecOutput("ip", "route", "show", "table", routeTable)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(route).ToNot(ContainSubstring(fmt.Sprintf("dev %s  scope link", wireguardInterfaceNameDefault)))
+			}
+		})
+
+		It("v3 node resource doesn't contain public-keys", func() {
+			for _, felix := range felixes {
+				Eventually(func() string {
+					node, err := client.Nodes().Get(context.Background(), felix.Hostname, options.GetOptions{})
+					Expect(err).NotTo(HaveOccurred())
+					wgPubKey := node.Status.WireguardPublicKey
+					return wgPubKey
+				}, "2s", "100ms").Should(BeEmpty())
+			}
+		})
+
+		It("workload connectivity remains but uses un-encrypted tunnel", func() {
+			for _, felix := range felixes {
+				Eventually(func() string {
+					// No tunnel implies un-encrypted communication.
+					out, _ := felix.ExecOutput("wg")
+					return out
+				}, "10s", "100ms").Should(BeEmpty())
+			}
+		})
+	})
+})
+
+var _ = infrastructure.DatastoreDescribe("WireGuard-Unsupported", []apiconfig.DatastoreType{apiconfig.EtcdV3, apiconfig.Kubernetes}, func(getInfra infrastructure.InfraFactory) {
+	var (
+		infra   infrastructure.DatastoreInfra
+		felixes []*infrastructure.Felix
+	)
+
+	BeforeEach(func() {
+		// Setup a single node cluster.
+		const nodeCount = 1
+
+		infra = getInfra()
+		felixes, _ = infrastructure.StartNNodeTopology(nodeCount, wireguardTopologyOptions(), infra)
+
+		// Install a default profile that allows all ingress and egress, in the absence of any Policy.
+		infra.AddDefaultAllow()
+
+		// TEMP: Copy wg binary.
+		//felixes[0].Container.EnsureBinary("wg")
+
+		felixes[0].TriggerDelayedStart()
+	})
+
+	AfterEach(func() {
+		if CurrentGinkgoTestDescription().Failed {
+			felixes[0].Exec("ip", "link")
+			felixes[0].Exec("wg")
+		}
+
+		felixes[0].Stop()
+
+		if CurrentGinkgoTestDescription().Failed {
+			infra.DumpErrorData()
+		}
+		infra.Stop()
+	})
+
+	It("no Wireguard device exists", func() {
+		Eventually(func() string {
+			out, _ := felixes[0].ExecOutput("ip", "-d", "link", "show", wireguardInterfaceNameDefault)
+			return out
+		}, "10s", "100ms").Should(ContainSubstring("does not exist"))
+	})
+
+	It("no wg tool info exists", func() {
+		Eventually(func() string {
+			out, _ := felixes[0].ExecOutput("wg")
+			return out
+		}, "10s", "100ms").Should(BeEmpty())
+	})
+})
+
+func wireguardTopologyOptions() infrastructure.TopologyOptions {
+	topologyOptions := infrastructure.DefaultTopologyOptions()
+
+	// For attaching tcpdump.
+	topologyOptions.DelayFelixStart = true
```

Comment:
We also need to delay start to ensure the calico-node resource is created before felix is started.